coalib/testing/LocalBearTestHelper.py

```diff
@@ -26,20 +26,45 @@ def execute_bear(bear, *args, **kwargs):
     return list(bear_output_generator)
 
 
+def get_local_bear_run_settings_dict(local_bear, settings_dict):
+    local_bear_metadata = local_bear.get_metadata()
+    compat_settings_dict = {key: settings_dict[key] for key in settings_dict
+                            if key in local_bear_metadata.optional_params}
+    return compat_settings_dict
+
+
 def get_results(local_bear,
                 lines,
                 filename=None,
                 force_linebreaks=True,
                 create_tempfile=True,
                 tempfile_kwargs={},
                 settings={}):
+    deps_results = dict()
+    if local_bear.BEAR_DEPS:
+        # Get results of bear's dependencies first
+        for bear in local_bear.BEAR_DEPS:
+            uut = bear(local_bear.section, queue.Queue())
+            deps_results[bear.name] = get_results(uut,
+                                                  lines,
+                                                  filename,
+                                                  force_linebreaks,
+                                                  create_tempfile,
+                                                  tempfile_kwargs,
+                                                  settings)
+
     with prepare_file(lines, filename,
                       force_linebreaks=force_linebreaks,
                       create_tempfile=create_tempfile,
-                      tempfile_kwargs=tempfile_kwargs) as (file, fname), \
-        execute_bear(local_bear, fname, file,
```

Comment:
Well, when I try that, I got this:
```
E           AssertionError: Bear returned None on execution
E            
E           Running bear TestDepBearA...
E           Bear TestDepBearA failed to run on file /tmp/tmpuf5mq4w6. Take a look at debug messages (`-V`) for further information.
E           The bear TestDepBearA raised an exception. If you are the author of this bear, please make sure to catch all exceptions. If not and this error annoys you, you might want to get in contact with the author of this bear.
E           
E           Traceback information is provided below:
E           
E           Traceback (most recent call last):
E             File "/home/rafid/GitHub/coala/coalib/bears/Bear.py", line 292, in execute
E               result = self.run_bear_from_section(args, kwargs)
E             File "/home/rafid/GitHub/coala/coalib/bears/Bear.py", line 285, in run_bear_from_section
E               return self.run(*args, **kwargs)
E           TypeError: run() got an unexpected keyword argument 'dependency_results'
```

That's caused by run method of bear which doesn't have dependency doesn't have `dependency_results` arg.

> If that doesnt work, IMO Bear needs to be enhanced to detect kwarg dependency_results and re-route that argument so it works correctly.
Do we need to create an issue for this?